File: backend/app/routes/session_logs.py
"""Session log routes — mirrors backend/src/routes/session-logs.ts."""

from __future__ import annotations

from fastapi import APIRouter, Request
from fastapi.responses import JSONResponse

from ..deps import admin_authenticated
from ..services.session_log import list_session_logs, read_session_log

router = APIRouter()


@router.get("")
async def list_logs(request: Request) -> dict:
    if not admin_authenticated(request):
        return JSONResponse({"error": "Not authenticated"}, status_code=401)
    try:
        return {"logs": await list_session_logs()}
    except Exception as e:  # noqa: BLE001
        return JSONResponse({"error": str(e) or "Failed"}, status_code=500)


@router.get("/{run_id}")
async def read_log(run_id: str, request: Request) -> dict:
    if not admin_authenticated(request):
        return JSONResponse({"error": "Not authenticated"}, status_code=401)
    try:
        events = await read_session_log(run_id)
        return {"runId": run_id, "events": events}
    except Exception as e:  # noqa: BLE001
        return JSONResponse({"error": str(e) or "Failed"}, status_code=500)

File: backend-legacy/src/routes/session-logs.ts
import { Hono } from 'hono';
import { listSessionLogs, readSessionLog } from '../services/session-log';

const sessionLogs = new Hono();

// List all session logs (most recent first)
sessionLogs.get('/', async (c) => {
  if (!c.get('auth').authenticated) {
    return c.json({ error: 'Not authenticated' }, 401);
  }
  try {
    const logs = await listSessionLogs();
    return c.json({ logs });
  } catch (e) {
    return c.json({ error: e instanceof Error ? e.message : 'Failed' }, 500);
  }
});

// Read a specific session log
sessionLogs.get('/:runId', async (c) => {
  if (!c.get('auth').authenticated) {
    return c.json({ error: 'Not authenticated' }, 401);
  }
  const runId = c.req.param('runId');
  try {
    const events = await readSessionLog(runId);
    return c.json({ runId, events });
  } catch (e) {
    return c.json({ error: e instanceof Error ? e.message : 'Failed' }, 500);
  }
});

export default sessionLogs;
